sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -814,6 +814,27 @@ build(const kernel_bundle<bundle_state::input> &InputBundle,
 
 namespace ext::oneapi::experimental {
 
+/////////////////////////
+// PropertyT syclex::include_files
+/////////////////////////
+struct include_files
+    : detail::run_time_property_key<detail::PropKind::IncludeFiles> {
+  include_files();
+  include_files(const std::string &name, const std::string &content) {
+    record.emplace_back(std::make_pair(name, content));
+  }
+  void add(const std::string &name, const std::string &content) {
+    record.emplace_back(std::make_pair(name, content));
+  }
+  std::vector<std::pair<std::string, std::string>> record;
+};
+using include_files_key = include_files;
+
+template <>
+struct is_property_key_of<include_files_key,
+                          sycl::kernel_bundle<bundle_state::ext_oneapi_source>>
```

Comment:
This specialization of `is_property_key_of` is not part of the specified API.

I think you are using this as a constraint for both `create_kernel_bundle_from_source` and for the `build` overload that takes a source bundle.  This is not quite right because it means that any of the following properties can be used with either of those functions:

* `include_files_key`
* `build_options_key`
* `save_log_key`
* `registered_kernel_names_key`

However, the correct behavior is to allow only `include_files_key` for `create_kernel_bundle_from_source` and to allow only the others for `build`.

Can we instead do something like this:

* Create a couple of internal types like:

  ```
  namespace detail {
  struct create_bundle_from_source_props;
  struct build_source_bundle_props;
  }
  ```

* Specialize `is_property_key_of` on these internal types.
* Update the constraints for `create_kernel_bundle_from_source` and `build` as appropriate.

Reply:
yeah, looks like I copy/pasted the key support over. Fixed.